sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -529,11 +574,33 @@ class GpuDynamicPartitionDataSingleWriter(
             }
           }
           // write concat table
-          currentWriterStatus.outputWriter.write(concat, statsTrackers)
-          currentWriterStatus.recordsInFile += concat.numRows
+          if (!needSplitBatch(
+            maxRecordsPerFile,
+            currentWriterStatus.recordsInFile,
+            concat.numRows))
+          {
+            statsTrackers.foreach(_.newBatch(concat))
+            currentWriterStatus.recordsInFile += concat.numRows()
+            currentWriterStatus.outputWriter.write(concat, statsTrackers)
+          } else {
+            withResource(concat){concat =>  
+              writeBatch(concat)
+            }
+          }
         } else {
-          currentWriterStatus.outputWriter.write(batch, statsTrackers)
-          currentWriterStatus.recordsInFile += batch.numRows
+          if (!needSplitBatch(
+            maxRecordsPerFile,
+            currentWriterStatus.recordsInFile,
+            batch.numRows))
+          {
```

Comment:
style nit: we tend to have the `{` on the same line as the preceding with a space before it.

```
batch.numRows)) {
```